account/init_tenant.go

```diff
@@ -106,43 +110,52 @@ func CleanTenant(ctx context.Context, config tenantConfig, remove bool) error {
 }
 
 // ShowTenant fetches the current tenant state.
-func ShowTenant(ctx context.Context, config tenantConfig) (*tenant.TenantSingle, error) {
-
-	c, err := createClient(ctx, config)
+func ShowTenant(ctx context.Context, config tenantConfig, options ...configuration.HTTPClientOption) (*tenant.TenantSingle, error) {
+	c, err := createClient(ctx, config, options...)
 	if err != nil {
 		return nil, err
 	}
-
 	res, err := c.ShowTenant(goasupport.ForwardContextRequestID(ctx), tenant.ShowTenantPath())
 	if err != nil {
 		return nil, err
 	}
 	defer rest.CloseResponse(res)
-
-	if res.StatusCode == http.StatusOK {
+	switch res.StatusCode {
+	case http.StatusOK:
 		tenant, err := c.DecodeTenantSingle(res)
 		if err != nil {
 			return nil, errors.NewInternalError(ctx, err)
 		}
 		return tenant, nil
-	} else if res.StatusCode > 400 {
+	case http.StatusNotFound:
```

Comment:
In scenarios where api.openshift.io is acting as a gateway/proxy, I've tried to directly convert the response code to a known WIT error type in a more uniform way ( sure there could be a better way to do it )

https://github.com/fabric8-services/fabric8-wit/commit/f704d8cbd38e1ef04e923a0127cc176f96e165c6#diff-ebe2c82ed83d49b7c1b2c79e7fd306f1R141

Instead of having the switch-case here for non-200, I suggest you pass it to a re-usable method ( as above ) which does the conversion for all non-200 cases.